libpod/stats.go

```diff
@@ -0,0 +1,115 @@
+package libpod
+
+import (
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/containerd/cgroups"
+	"github.com/opencontainers/runc/libcontainer"
+	"github.com/pkg/errors"
+)
+
+// ContainerStats contains the statistics information for a running container
+type ContainerStats struct {
+	ContainerID string
+	CPU         float64
+	CPUNano     uint64
+	SystemNano  uint64
+	MemUsage    uint64
+	MemLimit    uint64
+	MemPerc     float64
+	NetInput    uint64
+	NetOutput   uint64
+	BlockInput  uint64
+	BlockOutput uint64
+	PIDs        uint64
+}
+
+// GetContainerStats gets the running stats for a given container
+func (c *Container) GetContainerStats(previousStats *ContainerStats) (*ContainerStats, error) {
```

Comment:
Can you do a lock/defer unlock and syncContainer here to make sure it still exists? Probably better to get that error than a more generic "we can't find the cgroup"